### ISSUE TITLE
"Garage door" not working.

### ISSUE THREAD

**Issue by Jeanke** [Sinclair81/Homebridge-Logo-Platform#41]
Hello,

Some time ago I went from the LOGO TS plugin to this LOGO platform plugin.
I was able to get everything to work well again except for the garage door.
And for a while now I am running two plugins, one with the LOGO TS for the garage door and the other one with LOGO PLATFORM for all the rest.
Today I have been testing again to also make the garage door work on the logo platform plugin, but until now no succes.

For my garage door I have 2 ways to open it.
- One with an external remote that sends via a logo output a pulse signal tot the open/close input (input 1 of my garage door) of my garage door.
- Second way is wit the homebridge plugin. With the "LOGO TS" plugin I was able to use 2 logo outputs to go to the separate "open" and "close" inputs of my garage door (input 2 and 3of my garage door)

It seems like this last way is not possible anymore, or am I wrong?
What is the functionality exactly of the "garage door" knowing that I only use digital in and outputs?

I was able to make the "garage door" work on its own, but once I open ore close it in between with the remote the operation with homebridge seems to be comfused because the state of the garage door is not what it expects anymore.
The state of my garage door is also given by the garage door itselve, so I can not change this.

Before
 "accessory": "Logo-TS",
    "name": "Poort inrit",
    "interface": "modbus",
    "ip": "X.X.X.X",
    "port": XXX,
    "updateInterval": 1000,
    "type": "garagedoor",
    "garagedoorState": "M1",
    "garagedoorOpen": "M2",
    "garagedoorClose": "M3"

Now
  "platform": "LogoPlatform",
            "name": "Logo Poort",
            "interface": "modbus",
            "ip": "X.X.X.X,
            "port": XXX,
            "debugMsgLog": 0,
            "updateInterval": 1000,
            "pushButton": 1,
            "devices": [
                {
                    "name": "Poort inrit",
                    "type": "garagedoor",
                    "garagedoorGetState": "M1",
                    "garagedoorGetTargetState": "M1",
                    "garagedoorSetTargetState": "VW71"

I tought in my case that for VW71 I can make V71.0 to open and V71.1 to close, but only 71.0 seems to change state or am I doing something wrong.

Can you maybe help me with an example or som more info?

I hope my explenation is a little bit clear. :-)
Thanks in advance.

**Comment by Sinclair81** [comment]
Hi,
how exactly does it work with the remote control?
Remote control -> LOGO! Digital input -> LOGO! Program -> LOGO! Digital output -> Gate control ??
Only one input and only one output?

And you would like to use 2 outputs of the LOGO! from the Homebridge?
So that open and closed cannot be confused?

Is there a limit switch on the gate so that the LOGO! can determine whether the gate is open or closed?
Probably not?

The new plugin doesn't care whether the inputs are digital or analogue.
They just have to be specified correctly. V... for digital or VW... for analogue.
Only the output has to be analogue!
But I have built a UDF for this which converts the analogue value into digital.

Please look at the pictures in the LOGO! folder.

To stick with your settings:
They are correct, only in the LOGO! my garage UDF (LOGO!/Garage.lma) must come after the analog network input for VW71 and then all functions can be controlled with it.

But if you can't evaluate the status of the gate with the LOGO!, you can only do the opposite of what M1 returns! So if M1 tells the Homebridge that the gate is closed, you can only open it and vice versa.

So you have to make sure that the LOGO! always changes the status of M1 when you have used the remote control or the plug-in.

I hope you understand what I mean.

If you are not sure or it still doesn't work, you can also send me a picture of the LOGO! program part that controls the garage door by email.

**Comment by Jeanke** [comment]
Hello,
Thanks for the quick and extensive answer.

The remote control is as you describe it and goes to the open/close input of the gate controll.
The gate also has a built in limit switch that is connected to the LOGO! input that just puts it directly on M1.
So all the time that the gate is open M1=1, and when the gate is competely closed and stopped, M1=0. From the moment the gate starts moving to open M1 becomes 1 again.

Next to the "open/close" pulse input on the gate control I also have a separate "open" input and a separate "close" input that I indeed would like to control with two outputs of the LOGO!/Homebridge.

I looked at your example and implemented the UDF into my program with only difference that the open and close in my program go to sepparate outputs of the LOGO! for opening and closing, but still I am getting stuck at some point and most of the time can not open the gate anymore with Homebridge.

Going to do some more tests tomorrow, but out of ideas for today.

**Comment by Sinclair81** [comment]
OK, it could be the Home app.
If I have the Homebridge web interface and the Home app open at the same time.
I have noticed that the Homebridge always displays all the states of the dimmable lamps and blinds correctly.
The Home app never updates its view, though.

Does opening and closing the garage always work in the Homebridge?
Even if it doesn't work in the Home app?

**Comment by Jeanke** [comment]
Hello,
Sorry for the late answer, I had not much time to play wit it. :-)
In the end I made it work until now.
I mostly implemented your UDF and example in my program, but only thing that did not work in my case was the VW38 on NAQ2 of your example. In a way this blocked opening the door again.
I just let this out and now it works.
Maybe it has something to do with the fact that when closing the door in my case the feedback of "door closed" only comes when the door is completely closed and stopped.

Only thing that I also saw when testing but does not give a problem at the moment, is that when the door is closed and the feedback becomes 0, I get another pulse for closing out of the UDF.

Thanks for the help already.
